### DISCUSSION TITLE
Does it support ESP-IDF without arduino now?

### DISCUSSION THREAD

**Discussion by rqi14** [gin66/FastAccelStepper#217] [Q&A]
Hi. I found there has been a discussion regarding the support for ESP-IDF. But it seems unclear whether arduino (in component form) is still needed. Several years has passed since the forementioned discussion. I am wondering if there has been any change since then. Does it support ESP-IDF without arduino now?

**Comment by gin66** [comment]
As I am not an Esp-idf user, I would be willing to debug an existing sample project. I just do not intend to dive into the esp-idf configuration setup.

**Reply by freelancer1845** [reply]
I only put together a quick test, but for me the following worked on **ESP-IDF v5.3.1** and **ESP32-S3**

1. **idf_compoent.yml**
Remove dependencies (i. e. empty file)
3. **CMakeLists.txt**
Currently it REQUIRES *arduino-esp32* and also *driver* + *soc*. But the only drivers actually required are **esp_driver_gpio**, **esp_driver_rmt** and **esp_driver_pcnt**.
So the file changes to:
```
idf_component_register(
    SRC_DIRS "src"
    INCLUDE_DIRS "src"
    REQUIRES 
        esp_driver_gpio
        esp_driver_rmt
        esp_driver_pcnt
)
```

I don't know about the **mcpwm** but since support after IDF v5 is dropped anyways, that shouldn't be a problem.

The drivers cannot go into PRIV_REQUIRES because the headers are included in the hardware abstraction layer which in turn is included into "FastAccelStepper.h".

Currently I don't have an Axis at hand to really test the changes. If anyone is willing to assist and create a Pull Request, feel free :)

**Reply by gin66** [reply]
For a while, there is already an espidf project without arduino.
It is located here: extras/idf_project_example

This is always build as github [action](https://github.com/gin66/FastAccelStepper/actions/workflows/build_examples_esp_idf_project.yml)

**Reply by freelancer1845** [reply]
Yes but it includes "espressif/arduino-esp32" which in turn includes all the arduino specific things into the build.
For a project that doesn't want to use arduino this an unnecessary dependency. 
For instance, I was unable to compile my project because of a redefinition deep into to LwIP stack. Maybe they already fixed that, but why include arduino if it isn't required?

**Reply by gin66** [reply]
I am not an espidf expert. I am using platformio. And from the definition, arduino is not included:
```
[env:esp32_idf_V6_10_0]
platform    = espressif32 @ 6.10.0
board       = esp32dev
framework   = espidf
build_flags = -Werror -Wall -Wno-deprecated-declarations -Wno-error=incompatible-pointer-types -Wno-unused-function -Wno-error=attributes -DCONFIG_RMT_ENABLE_DEBUG_LOG=5
board_build.f_cpu = 240000000L
lib_extra_dirs = ../..
```

Could you please guide, where this inclusion is done?

**Reply by freelancer1845** [reply]
It is included from "idf_component.yml"
```
dependencies:
  espressif/arduino-esp32:
    version: "*"
```
But it could be necessary for the platformio build if one wants to use "framework = arduino", I'm not sure.

**Reply by gin66** [reply]
I will try later to remove this. Will see if the build breaks then….

**Reply by gin66** [reply]
ok. the build breaks and I have no idea, where the remaining reference to arduino-esp32 is…

**Reply by gin66** [reply]
found it.

**Reply by gin66** [reply]
successfully built. Guess now without arduino-esp32

**Reply by gin66** [reply]
I should have read your post more carefully. I just repeated your steps with much more time….

**Reply by gin66** [reply]
only the last step in regard to soc is not done…

**Reply by freelancer1845** [reply]
Ok, let me know if there is anything more I can help with :)
Thanks a lot for the library!
